x/staking/keeper/validator.go

```diff
@@ -434,15 +432,15 @@ func (k Keeper) UnbondAllMatureValidators(ctx sdk.Context) {
 				if !val.IsUnbonding() {
 					panic("unexpected validator in unbonding queue; status was not unbonding")
 				}
-				if !val.UnbondingOnHold {
+				if val.UnbondingOnHoldRefCount == 0 {
```

Comment:
Isn't .OnHold() doing this elsewhere?

Reply:
`OnHold()` is defined for unbonding delegation entries and redelegation entries. It is useful there since we use it in multiple places in the code, similarly to the `IsMature()`. For validators, there is no `IsMature()`. Also, this is the only place we check if `val.UnbondingOnHoldRefCount == 0`.